Sources/Amplitude/Amplitude.m

```diff
@@ -284,6 +280,82 @@ - (instancetype)initWithInstanceName:(NSString *)instanceName {
     return self;
 }
 
++ (NSString *)getDataStorageKey:(NSString *)key {
+    NSString *dataStorageKey = [NSString stringWithFormat:@"%s_%@_%@_%@", "amplitude", _bundleIdentifier, self.instanceName, key];
+    return dataStorageKey;
+}
+
+- (void)migrateToFileStorage {
+    NSString *fileStoragePath =  [AMPStorage getAppStorageAmpDir:self.instanceName];
+    BOOL isDir;
+    [[NSFileManager defaultManager] fileExistsAtPath:fileStoragePath isDirectory:&isDir];
+    if (isDir) return;
+
+    NSString *databaseDirectory = [AMPUtils platformDataDirectory];
+    NSString *databasePath = [databaseDirectory stringByAppendingPathComponent:@"com.amplitude.database"];
+    if (![self->_instanceName isEqualToString:kAMPDefaultInstance]) {
+        databasePath = [NSString stringWithFormat:@"%@_%@", databasePath, self.instanceName];
+    }
+    BOOL hasDatabase =  [[NSFileManager defaultManager] fileExistsAtPath:databasePath];
+    if (!hasDatabase) return;
+
+    int eventCount = [self.dbHelper getEventCount];
+    if (eventCount > 0) {
+        NSMutableArray *events = [self.dbHelper getEvents: -1 limit: -1];
+        self->_eventsBuffer = [events mutableCopy];
+        for(NSMutableDictionary *event in events) {
+            NSError *error = nil;
+            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:[AMPUtils makeJSONSerializable:event] options:0 error:&error];
+            if (error != nil) {
+                AMPLITUDE_ERROR(@"ERROR: could not JSONSerialize event type: %@", error);
+                continue;
+            }
+            NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+            [AMPStorage storeEvent:jsonString instanceName:self.instanceName];
+        }
+       NSDictionary *eventsDict = [self mergeEventsAndIdentifys:events identifys:nil numEvents:eventCount];
```

Comment:
Why are we merging here if we're doing the identify events separately on the next few lines (321)?

Reply:
Good catch. You are right, we only need a maxEventId here for removing all events in the database.  Will fix this.